packages/components/src/mobile/bottom-sheet/index.native.js

```diff
@@ -89,16 +96,66 @@ class BottomSheet extends Component {
 		Dimensions.addEventListener( 'change', this.onDimensionsChange );
 	}
 
-	keyboardWillShow( e ) {
+	keyboardShow( e ) {
+		if ( ! this.props.isVisible ) {
+			return;
+		}
+
 		const { height } = e.endCoordinates;
+		this.keyboardHeight = height;
+		this.performKeyboardLayoutAnimation( e );
+		this.onSetMaxHeight();
+		this.props.onKeyboardShow?.();
```

Comment:
This callback and `onKeyboardHide` will allow us to sync potential changes in the layout with the layout animations triggered from `performKeyboardLayoutAnimation`.